client/state/reader/teams/reducer.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * External Dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	READER_TEAMS_REQUEST,
+	READER_TEAMS_REQUEST_FAILURE,
+	READER_TEAMS_REQUEST_SUCCESS,
+} from 'state/action-types';
+import { createReducer } from 'state/utils';
+
+export const items = createReducer( [], {
+	[ READER_TEAMS_REQUEST_SUCCESS ]: ( state, action ) => [ ...state, ...action.teams ],
+} );
+
+export const isRequesting = createReducer( false, {
+	[ READER_TEAMS_REQUEST ]: () => true,
+	[ READER_TEAMS_REQUEST_FAILURE ]: () => false,
+	[ READER_TEAMS_REQUEST_SUCCESS ]: () => false,
```

Comment:
One thing to think about when moving towards this API middleware approach is whether or not we need to store state specific to things like `isLoading`. Is it possible that we can represent a loading state without a separate boolean and without needing to send around `_SUCCESS` and `_FAILURE` actions.

is it possible that a `null` value for the `items` state implies that we haven't yet loaded the data whereas an empty list could indicate that the data has loaded yet there are no items in it?

much of the `isRequesting` type values aren't really required to control the display of what's going on in Calypso and if we can live without them we ought to strive for that. a minimal Redux tree is simpler, faster, and clearer in what it's trying to convey.